.github/workflows/release.yml

```diff
@@ -29,3 +29,13 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Publish dev snapshot
+        if: steps.changesets.outputs.published != 'true'
+        run: |
+          git checkout main
+          pnpm changeset version --snapshot dev
+          pnpm changeset publish --tag dev
```

Comment:
changesets/action has no support of snapshots (https://github.com/changesets/action/issues/150) so doing manually.